### PR TITLE
Created PageObjectFactory class

### PR DESCRIPTION
**Pull request by eduardomcjl**
Created page object factory class and use it in login and boards test cases

### CODE REVIEW

File: main/trello/gui/page_objects/PageObjectFactory.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+    def __init__(self):
+        self.factory = {
+            "login": LoginPage,
+            "board": CardsPage,
+            "home": HomePage
+        }
+
+    def get_page(self, page_name):
+        return self.factory[page_name]
```
Comment (sebasonti-fundjala) [inline main/trello/gui/page_objects/PageObjectFactory.py:16]: You can receive the driver as a parameter in the constructor and instantiate the Page Object right here. The driver would be passed on to the PO factory on the environment.py file

File: tests/gui/environment.py [hunk 2]
```diff
@@ -28,6 +29,8 @@ def before_scenario(context, scenario):
     context.driver = webdriver.Chrome(
         service=chrome_service, options=chrome_options)
 
+    context.page_factory = PageObjectFactory()
```
Comment (sebasonti-fundjala) [inline tests/gui/environment.py:32]: You can pass the driver to the PO factory right here, something like this:
PageObjectFactory(context.driver)

File: tests/gui/boards.feature [hunk 3]
```diff
@@ -2,7 +2,7 @@ Feature: Trello UI Boards
     As a User I want to be able to make CRUD and other operations with boards
 
     Background: The user is logged in
-        Given the user goes to "/login"
+        Given the user goes to "login" page
```
Comment (sebasonti-fundjala) [inline tests/gui/boards.feature:5]: This should be made only in one step, "Given the user is logged in" and the navigation to the /login path should be made on the step implementation

File: tests/gui/steps/common_steps.py [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
-from main.trello.gui.page_objects.login_page import LoginPage
 
 
-@given('the user goes to "{page}"')
+@given('the user goes to "{page}" page')
 def step_impl(context, page):
-    context.driver.get(context.base_url + page)
+    context.login_page = context.page_factory.get_page(page)(context.driver)
+    if page == "login":
+        context.driver.get(context.base_url + '/login')
```
Comment (sebasonti-fundjala) [inline tests/gui/steps/common_steps.py:8]: not sure if this is better, you could have left just "/login", no need to say "login page"

**Review by sebasonti-fundjala — CHANGES_REQUESTED** [review]

**Review by sebasonti-fundjala — APPROVED** [review]

**Review by jorgeposterla — APPROVED** [review]

**Pull request merged**